plugins/services/src/js/components/ConfigurationMapTable.js

```diff
@@ -0,0 +1,380 @@
+ *         return row.firstKey;
+ *       },
+ *       (row, rowIndex) => {
+ *         return row.secondKey;
+ *       },
+ *       (row, rowIndex) => {
+ *         return <strong>row.thirdKey</strong>;
+ *       }
+ *     ]}
+ *     columns={(dataKeys, data) => {
+ *       return [
+ *         (row, rowIndex) => row.firstKey,
+ *         (row, rowIndex) => row.secondKey,
+ *         (row, rowIndex) => row.thirdKey,
+ *       ];
+ *     }}
+ *
+ *     //
+ *     // Specify the empty value placeholder
+ *     //
+ *     emptyPlaceholder={<span>&mdash;</span>}
+ *     emptyPlaceholder={['-', '(None)', '(None)']}
+ *     emptyPlaceholder={(columnIndex, row, rowIndex) => {
+ *        if (columnIndex === 0) {
+ *          return <span>&mdash;</span>;
+ *        }
+ *
+ *        return '(None)';
+ *     }}
+ *
+ *     //
+ *     // You can also specify a custom renderer, but use this only if:
+ *     //
+ *     // (1) You want all cells wrapped with some renderer
+ *     // (2) You are using key-based `columns` and you already have the render
+ *     //     functions for each key.
+ *     //
+ *     // Otherwise it might be simpler to use a `column` function and return
+ *     // the rendererd value directly.
+ *     //
+ *     renderWith={(value, columnIndex, row, rowIndex) => {
+ *        return <pre>{value}</pre>;
+ *     }}
+ *     renderWith={[
+ *        (value, columnIndex, row, rowIndex) => <pre>{value}</pre>,
+ *        (value, columnIndex, row, rowIndex) => <span>{value}</span>
+ *        (value, columnIndex, row, rowIndex) => <strong>{value}</strong>
+ *     }}
+ *     renderWith={[
+ *        (value, columnIndex, row, rowIndex) => <pre>{value}</pre>,
+ *        <span>Static value</span>,
+ *        <span>Another static value</span>
+ *     }}
+ *
+ *     //
+ *     // Hide columns following some particular criteria
+ *     //
+ *     hideColumns="never"
+ *     hideColumns="empty"
+ *     hideColumns={[
+ *       (rowValues, data) => return rowValues.length === 0,
+ *       (rowValues, data) => return data.someFlag != null;
+ *     ]}
+ *     hideColumns={(rowValues, data) => return rowValues.length === 0}
+ *
+ *     //
+ *     // You can either pass an array, or an object in which case you will be
+ *     // iterating over it's key values in an array manner.
+ *     //
+ *     // When an object is given, you can find the current object key as the
+ *     // `rowIndex` in the functions above.
+ *     //
+ *     data={anArrayOfObjects}
+ *     data={anObject}
+ *
+ *   />
+ */
+class ConfigurationMapTable extends React.Component {
+
+  getRowValuesFor(data, dataKeys, column) {
+    let values = [];
+
+    for (let i=0, length=dataKeys.length; i<length; ++i) {
+      let row = data[dataKeys[i]];
+      let value = null;
+
+      if (typeof column === 'function') {
+        value = column(row, propKey);
+      } else {
+        value = row[column];
+      }
+
+      values.push(value);
+    }
+
+    return values;
+  }
+
+  render() {
```

Comment:
There is a lot going on in this render function. Can we split into separate functions, to make it more readable?